packages/appstore/src/components/modals/account-type-modal/account-type-modal.tsx

```diff
@@ -92,18 +110,40 @@ const MT5AccountTypeModal = () => {
     const { trading_platform_available_accounts } = client;
     const { enableApp, disableApp } = ui;
     const { setAppstorePlatform } = common;
+
+    React.useEffect(() => {
+        if (!is_account_type_modal_visible) {
+            selectAccountTypeCard('');
+        }
+    }, [is_account_type_modal_visible, selectAccountTypeCard]);
+
     const is_financial_available = trading_platform_available_accounts.some(
         (available_account: TTradingPlatformAvailableAccount) => available_account.market_type === 'financial'
     );
 
     const is_synthetic_available = trading_platform_available_accounts.some(
         (available_account: TTradingPlatformAvailableAccount) => available_account.market_type === 'gaming'
     );
+    const is_swapfree_available = trading_platform_available_accounts.some(
+        (available_account: TTradingPlatformAvailableAccount) => available_account.market_type === 'all'
+    );
```

Comment:
Can you extract this into a hook ? 
and also do a test for it 😸 

cc: @farzin-deriv

Reply:
@thisyahlen-deriv Nice 😂

Reply:
We can revamp this as we have to revamp the financial and synthetic as well and make a hook for all three of them

Reply:
Done